op-acceptance-tests/tests/sync_tester/hardforks_ext/hardforks_ext.go

```diff
@@ -0,0 +1,199 @@
+package hardforks_ext
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/ethereum/go-ethereum/log"
+
+	"github.com/ethereum-optimism/optimism/op-devstack/devtest"
+	"github.com/ethereum-optimism/optimism/op-devstack/dsl"
+	"github.com/ethereum-optimism/optimism/op-devstack/shim"
+	"github.com/ethereum-optimism/optimism/op-devstack/stack"
+	"github.com/ethereum-optimism/optimism/op-devstack/stack/match"
+	"github.com/ethereum-optimism/optimism/op-devstack/sysgo"
+	"github.com/ethereum-optimism/optimism/op-service/eth"
+	"github.com/ethereum-optimism/optimism/op-service/retry"
+	"github.com/ethereum-optimism/optimism/op-service/testlog"
+	"github.com/ethereum-optimism/optimism/op-supervisor/supervisor/types"
+)
+
+// Configuration defaults for op-sepolia
+const (
+	DefaultL2NetworkName      = "op-sepolia"
+	DefaultL1ChainID          = 11155111
+	DefaultL2ELEndpoint       = "https://ci-sepolia-l2.optimism.io"
+	DefaultL1CLBeaconEndpoint = "https://ci-sepolia-beacon.optimism.io"
+	DefaultL1ELEndpoint       = "https://ci-sepolia-l1.optimism.io"
+
+	// Tailscale networking endpoints
+	DefaultL2ELEndpointTailscale       = "https://proxyd-l2-sepolia.primary.client.dev.oplabs.cloud"
+	DefaultL1CLBeaconEndpointTailscale = "https://beacon-api-proxy-sepolia.primary.client.dev.oplabs.cloud"
+	DefaultL1ELEndpointTailscale       = "https://proxyd-l1-sepolia.primary.client.dev.oplabs.cloud"
+)
+
+var (
+	// Network upgrade block numbers for op-sepolia
+	networkUpgradeBlocks = map[string]uint64{
+		"Canyon":   4089330,
+		"Delta":    5700330,
+		"Ecotone":  8366130,
+		"Fjord":    12597930,
+		"Granite":  15837930,
+		"Holocene": 20415330,
+		"Isthmus":  26551530,
+	}
+
+	// Load configuration from environment variables with defaults
+	L2NetworkName = getEnvOrDefault("L2_NETWORK_NAME", DefaultL2NetworkName)
+	L1ChainID     = eth.ChainIDFromUInt64(getEnvUint64OrDefault("L1_CHAIN_ID", DefaultL1ChainID))
+
+	// Default endpoints
+	L2ELEndpoint       = getEnvOrDefault("L2_EL_ENDPOINT", DefaultL2ELEndpoint)
+	L1CLBeaconEndpoint = getEnvOrDefault("L1_CL_BEACON_ENDPOINT", DefaultL1CLBeaconEndpoint)
+	L1ELEndpoint       = getEnvOrDefault("L1_EL_ENDPOINT", DefaultL1ELEndpoint)
+)
+
+// getEnvOrDefault returns the environment variable value or the default if not set
+func getEnvOrDefault(envVar, defaultValue string) string {
+	if value := os.Getenv(envVar); value != "" {
+		return value
+	}
+	return defaultValue
+}
+
+// getEnvUint64OrDefault returns the environment variable value as uint64 or the default if not set
+func getEnvUint64OrDefault(envVar string, defaultValue uint64) uint64 {
+	if value := os.Getenv(envVar); value != "" {
+		if parsed, err := strconv.ParseUint(value, 10, 64); err == nil {
+			return parsed
+		}
+	}
+	return defaultValue
+}
+
+// setupOrchestrator initializes and configures the orchestrator for the test
+func setupOrchestrator(gt *testing.T, t devtest.T, blk uint64) *sysgo.Orchestrator {
```

Comment:
This orchestration part may be moved to `op-devstack/presets/orchestrator.go` and more abstracted.

Reply:
Same comment as above:

I'd rather move it only once we use it at more than one place. At the moment it is very much coupled with these tests, so I'd rather it lives in proximity to them.

As soon as we have some other tests that uses similar orchestration, and abstraction makes sense, we can do it.